firebase-perf/src/test/java/com/google/firebase/perf/application/FragmentStateMonitorTest.java

```diff
@@ -169,116 +165,101 @@ public void lifecycleCallbacks_onPausedCalledBeforeOnResume_doesNotLogFragmentSc
   public void
       lifecycleCallbacks_differentFrameMetricsCapturedByFma_logFragmentScreenTraceWithCorrectFrames() {
     FragmentStateMonitor monitor =
-        new FragmentStateMonitor(clock, mockTransportManager, appStateMonitor, fma);
-    when(fma.getMetrics()).thenReturn(fmaMetrics1);
+        new FragmentStateMonitor(clock, mockTransportManager, appStateMonitor, recorder);
+    doReturn(Optional.of(frameCounts1)).when(recorder).stopFragment(any());
     monitor.onFragmentResumed(mockFragmentManager, mockFragment);
     verify(mockTransportManager, times(0)).log(any(TraceMetric.class), any());
 
-    when(fma.getMetrics()).thenReturn(fmaMetrics2);
     monitor.onFragmentPaused(mockFragmentManager, mockFragment);
 
-    // fmaMetrics1 has 1+3+1=5 total frames, 3+1=4 slow frames, and 1 frozen frames
-    // fmaMetrics2 has 5+5+4=14 total frames, 5+4=9 slow frames, and 4 frozen frames
-    // we expect the trace to have 14-5=9 total frames, 9-4=5 slow frames, and 4-1=3 frozen
-    // frames.
     verify(mockTransportManager, times(1))
         .log(argTraceMetric.capture(), nullable(ApplicationProcessState.class));
     TraceMetric metric = argTraceMetric.getValue();
     Assert.assertEquals(
-        9, (long) metric.getCountersMap().get(Constants.CounterNames.FRAMES_TOTAL.toString()));
+        frameCounts1.getTotalFrames(),
+        (long) metric.getCountersMap().get(Constants.CounterNames.FRAMES_TOTAL.toString()));
     Assert.assertEquals(
-        5, (long) metric.getCountersMap().get(Constants.CounterNames.FRAMES_SLOW.toString()));
+        frameCounts1.getSlowFrames(),
+        (long) metric.getCountersMap().get(Constants.CounterNames.FRAMES_SLOW.toString()));
     Assert.assertEquals(
-        3, (long) metric.getCountersMap().get(Constants.CounterNames.FRAMES_FROZEN.toString()));
+        frameCounts1.getFrozenFrames(),
+        (long) metric.getCountersMap().get(Constants.CounterNames.FRAMES_FROZEN.toString()));
   }
 
+  /** Simulate call order of activity + fragment lifecycle events */
   @Test
   public void lifecycleCallbacks_cleansUpMap_duringActivityTransitions() {
-    // Simulate call order of activity + fragment lifecycle events
-    AppStateMonitor appStateMonitor =
-        spy(new AppStateMonitor(mockTransportManager, clock, configResolver, fma));
+    Bundle savedInstanceState = mock(Bundle.class);
+    AppStateMonitor appStateMonitor = mock(AppStateMonitor.class);
```

Comment:
Why do we need to create more mocks within a test? Can't we use the one created at the top with `@Mock`?

Reply:
Makes sense. Updated it.